src/cascade/model/priors.py

```diff
@@ -0,0 +1,178 @@
+from functools import total_ordering
+
+
+@total_ordering
+class _Prior:
+    """The base for all Priors
+    """
+
+    density = None
+
+    def __init__(self, name=None):
+        self.name = name
+
+    def _parameters(self):
+        raise NotImplementedError()
+
+    def parameters(self):
+        return dict(density=self.density, **self._parameters())
+
+    def __hash__(self):
+        return hash((frozenset(self.parameters().items()), self.name))
+
+    def __eq__(self, other):
+        return isinstance(other, _Prior) and self.name == other.name and self.parameters() == other.parameters()
+
+    def __lt__(self, other):
+        if not isinstance(other, _Prior):
+            return NotImplemented
+        return list(dict(name=self.name, **self.parameters()).items()) < list(
+            dict(name=other.name, **other.parameters()).items()
+        )
+
+    def __repr__(self):
+        return f"<{type(self).__name__} {self.parameters()}>"
+
+
+def _validate_bounds(lower, mean, upper):
+    if not lower <= mean <= upper:
+        raise ValueError("Bounds are inconsistent")
+
+
+def _validate_standard_deviation(standard_deviation):
+    if standard_deviation < 0:
+        raise ValueError("Standard deviation must be positive")
+
+
+def _validate_nu(nu):
+    if nu < 0:
+        raise ValueError("Nu must be positive")
+
+
+class UniformPrior(_Prior):
+    density = "uniform"
+
+    def __init__(self, lower, upper, mean=None, name=None):
+        super().__init__(name=name)
+        if mean is None:
+            mean = (upper - lower) / 2 + lower
```

Comment:
this is the same as (upper + lower) / 2 , just wondering about it

Reply:
It is. The other way made more sense to me at the time but now it just seems confusing. Yours is more clear.